libraries/stdlib/jvm/test/random/RandomSerializationTest.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package random
+
+import test.io.deserializeFromHex
+import test.io.serializeAndDeserialize
+import kotlin.random.Random
+import kotlin.random.asJavaRandom
+import kotlin.random.asKotlinRandom
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertSame
+
+class RandomSerializationTest {
+    @Test
+    fun defaultIsSerializable() {
+        val instance = Random
+        assertSame(instance, serializeAndDeserialize(instance))
+    }
+
+    private fun testPersistedDeserialization(hexValue: String, expected: Any) =
+        assertEquals(expected = expected, actual = deserializeFromHex(hexValue))
+
+    @Test
+    fun deserializeDefault() = testPersistedDeserialization(
```

Comment:
In this test we should use `assertSame` check. In the test with predictable Random, we can compare several `next*` values with the expected instance.

Reply:
How many of them?

Reply:
@ilya-g

Reply:
I believe one call for each of the following methods would be enough: `nextInt()`, `nextLong()`, `nextDouble()`, `nextBytes()`

Reply:
Clear